backend/src/main/scala/cromwell/backend/BackendWorkflowInitializationActor.scala

```diff
@@ -2,58 +2,116 @@ package cromwell.backend
+import cromwell.backend.validation.TryUtils
+import cromwell.backend.validation.exception.ValidationAggregatedException
+import cromwell.core._
+import lenthall.exception.AggregatedException
+import wdl4s.expression.{NoFunctions, ValueEvaluator}
+import wdl4s.{Call, WdlExpression}
 
 import scala.concurrent.Future
+import scalaz.{Failure, NonEmptyList}
 
 object BackendWorkflowInitializationActor {
 
   // Commands
   sealed trait BackendWorkflowInitializationActorCommand extends BackendWorkflowLifecycleActorCommand
-
   case object Initialize extends BackendWorkflowInitializationActorCommand
   final case class Abort(jobKey: BackendJobDescriptorKey) extends BackendWorkflowInitializationActorCommand
 
   // Responses
   sealed trait BackendWorkflowInitializationActorResponse extends BackendWorkflowLifecycleActorResponse
-
   sealed trait InitializationResponse extends BackendWorkflowInitializationActorResponse
   case object InitializationSuccess extends InitializationResponse
   case class InitializationFailed(reason: Throwable) extends Exception with InitializationResponse
+
 }
 
 /**
   * Workflow-level actor for executing, recovering and aborting jobs.
   */
 trait BackendWorkflowInitializationActor extends BackendWorkflowLifecycleActor with ActorLogging {
 
-  final def receive: Receive = LoggingReceive {
-    case Initialize    => performActionThenRespond(initSequence, onFailure = InitializationFailed)
-    case AbortWorkflow => performActionThenRespond(abortInitialization, onFailure = BackendWorkflowAbortFailedResponse)
+  def receive: Receive = LoggingReceive {
+    case Initialize => performActionThenRespond(initSequence(), onFailure = InitializationFailed)
+    case AbortWorkflow => performActionThenRespond(abortInitialization(), onFailure = BackendWorkflowAbortFailedResponse)
   }
 
   /**
     * Our predefined sequence to run during preStart
     */
-  final def initSequence = for {
-    _ <- validate
-    _ <- beforeAll
+  final def initSequence() = for {
+    _ <- validate()
+    _ <- beforeAll()
   } yield InitializationSuccess
 
   /**
     * Abort all initializations.
     */
-  def abortInitialization: Future[WorkflowAbortResponse]
+  def abortInitialization(): Future[WorkflowAbortResponse]
+
+  /**
+    * A call which happens before anything else runs
+    */
+  def beforeAll(): Future[Unit]
+
+  /**
+    * Validates runtime attributes for one specific call.
+    *
+    * @param runtimeAttributes Runtime Attributes with already evaluated values.
+    * @return If all entries from runtime attributes section are valid Success otherwise
+    *         Failure with the aggregation of errors.
+    */
+  def validateRuntimeAttributes(runtimeAttributes: EvaluatedRuntimeAttributes): ErrorOr[Unit]
 
   /**
     * Validate that this WorkflowBackendActor can run all of the calls that it's been assigned
     */
-  def validate: Future[Unit]
+  protected def validate(): Future[Unit] = {
+    Future {
```

Comment:
We should let the `validateRuntimeAttributes()` method return a Future and compose the response from there instead of wrapping everything in this method into a block Future.